### ISSUE TITLE
Search doesn't respect filter

### ISSUE THREAD

**Issue by SpacyRicochet** [insidegui/WWDC#254]
Searching through videos doesn't respect the filter that's currently active. This is related to #66, which presents a workaround.

However, if a filter is currently active, I think the better choice is to only search within the filtered items.  Currently it's very unclear that ignoring the filter is 'As Intended'.

Either that, or somehow indicate that filter is completely ignored and give you some tips on how to search more specifically.

![search_and_filter](https://user-images.githubusercontent.com/904596/27138448-c97f8aae-5120-11e7-9771-6571b1cfe429.gif)

**Comment by insidegui** [comment]
Can you elaborate?

What do you mean by "doesn't respect filter"?

**Comment by insidegui** [comment]
Interesting. It works for me. Filtering for a year and typing something only shows videos from the selected year.

![screen shot 2017-06-14 at 11 50 51](https://user-images.githubusercontent.com/67184/27138793-ba1c9972-50f7-11e7-97ef-4d80bc77025b.png)

**Comment by SpacyRicochet** [comment]
Ah, I thought this was going to be a feature request (hence my 'unintuitive' remark in the title). Looks like this is a bug instead. Let me know if you need more info, though I doubt I can get more detailed than the gif.

**Comment by insidegui** [comment]
@bcmn This happens to you too, right?

**Comment by bcmn** [comment]
No, I can't recreate this.

**Comment by SpacyRicochet** [comment]
Are there any magic files that I can delete (or store for posterity/this issue) to get rid of my state (don't mind starting from scratch)?

If it doesn't happen after that, it apparantly is some kind of weird state issue.

**Comment by bcmn** [comment]
I was playing with this again to try and narrow causes down, and _have_ now recreated it. I struggled to diagnose it closely, but did make a couple observations.

One possibility is that it may be related to the searching of the transcript for sessions? I noted that on searching "engineering", the 2015 ADAs are returned. On a cursory glance can't see the word "engineering" in any of the metadata, but it is present in the ASCIIwwdc transcript. It might be worth checking the JSON.

Another observation was that if I filter to 2016, instead of filtering to 2017 as per the original post, the 2017 engineering session disappears (though 2015 and earlier still show). However, it doesn't seem to be simply a boolean operation on year, as I attempted to recreate this with even earlier years, but 2016 sessions still showed. However, 2017 didn't show, which seemed to suggest it's handled differently somehow -- a notable difference between the years is that there are few (if any?) transcripts for 2017 sessions currently. 🤔
